overlord/state/state.go

```diff
@@ -379,14 +385,16 @@ func (s *State) Prune(pruneWait, abortWait time.Duration) {
 			}
 			continue
 		}
-		if readyTime.Before(pruneLimit) {
+		// change old or we have too many changes
+		if readyTime.Before(pruneLimit) || (chg.Status().Ready() && len(s.changes) > maxReadyChanges) {
```

Comment:
Nice catch, I removed the chg.Status().Ready() check now. 

Good catch on the logic, if 100 changes are in flight, this will always kill the one change that just became ready, not what we want! As a strawman I implemented a real maxReadyChanges that only counts the changes that are ready, not the total changes. Do you think that is reasonable? Or should we do something else instead (like a min-duration of e.g. 10min that we allow a change to stay even if the limit is reached?).